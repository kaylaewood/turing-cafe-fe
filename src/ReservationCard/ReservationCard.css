.card {
  background-color: white;
  border: 3px solid teal;
  border-radius: 10px;
  color: teal;
  padding: 1vw;
  margin: 2vw;
}

.cancel-reservation {
  width: 90%;
  margin: auto;
  background-color: teal;
  color: white;
  font-size: 1.3vw;
  padding: .3vw;
}
