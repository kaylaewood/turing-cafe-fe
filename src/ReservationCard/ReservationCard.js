import React from 'react';
import './ReservationCard.css';

const ReservationCard = ({ id, name, date, time, number, cancelReservation }) => {

  return(
    <div className='card'>
      <h2>{name}</h2>
      <p>{date}</p>
      <p>{time}</p>
      <p>Number of Guests: {number}</p>
      <button className='cancel-reservation' onClick={() => cancelReservation(id)}>Cancel</button>
    </div>
  )
}

export default ReservationCard;
