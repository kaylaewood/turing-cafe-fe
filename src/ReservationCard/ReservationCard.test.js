import React from 'react';
import { shallow } from 'enzyme';
import ReservationCard from './ReservationCard';

describe('ReservationCard', () => {
  let wrapper;
  let cancelReservationMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<ReservationCard
      id='1'
      name='Kayla'
      date='12/03'
      time='2:45'
      number='3'
      cancelReservation={cancelReservationMock}
    />)
  })

  it('should match snapshot as expected', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should invoke cancelReservation when button is clicked', () => {
    wrapper.find('button').simulate('click');
    expect(cancelReservationMock).toHaveBeenCalled();
  })
})
