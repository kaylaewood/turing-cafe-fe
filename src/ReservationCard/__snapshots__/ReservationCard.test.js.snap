// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservationCard should match snapshot as expected 1`] = `
<div
  className="card"
>
  <h2>
    Kayla
  </h2>
  <p>
    12/03
  </p>
  <p>
    2:45
  </p>
  <p>
    Number of Guests: 
    3
  </p>
  <button
    className="cancel-reservation"
    onClick={[Function]}
  >
    Cancel
  </button>
</div>
`;
